pkg/api/pod/util.go

```diff
@@ -658,24 +658,6 @@ func dropDisabledFields(
 		// For other types of containers, validateContainers will handle them.
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) && !rroInUse(oldPodSpec) {
```

Comment:
To achieve our [compatibility version](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md#compatibility-versions) promise, after three releases where a feature has been locked to the default value (whether that feature is GA or deprecated), feature gates and references should be removed. 

Please refer to  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md#ga-features.

Reply:
Fixed